containers/Dockerfile_arm64.buildenv

```diff
@@ -0,0 +1,388 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# If you change this file please push a new image to DockerHub so that the
+# new image is used. Docker must be run as root on your machine. There are 3
+# different images in this Dockerfile. Here is how to build them:
+# 1. `dev`
+#
+#   cd $SPECTRE_HOME
+#   docker build --target dev -t sxscollaboration/spectre_arm64:dev \
+#                 -f ./containers/Dockerfile_arm64.buildenv .
+#
+# 2. `ci` (this should use the `dev` image you just built for most of it)
+#
+#   cd $SPECTRE_HOME
+#   docker build --target ci -t sxscollaboration/spectre_arm64:ci \
+#                 -f ./containers/Dockerfile_arm64.buildenv .
+#
+# 3. `demo` 
+#
+#   cd $SPECTRE_HOME
+#   docker build --target demo -t sxscollaboration/spectre_arm64:demo \
+#                 -f ./containers/Dockerfile_arm64.buildenv .
+#
+#   docker push sxscollaboration/spectre_arm64:dev
+#   docker push sxscollaboration/spectre_arm64:ci
+#
+#    To build `demo`, you must be in $SPECTRE_ROOT and there cannot be a
+#    directory named `build` in $SPECTRE_ROOT because the image will create
+#    this directory (in the container).
+#
+#   cd $SPECTRE_HOME
+#   rm -rf build/
+#   docker build --target demo -t sxscollaboration/spectre_arm64:demo \
+#                 -f ./containers/Dockerfile_arm64.buildenv .
+#
+#    and then to push the `demo` image to DockerHub:
+#
+#   docker push sxscollaboration/spectre_arm64:demo
+#
+# If you do not have permission to push to DockerHub please coordinate with
+# someone who does. Since changes to this image effect our testing
+# infrastructure it is important all changes be carefully reviewed.
+
+FROM --platform=linux/arm64 ubuntu:20.04 AS dev
```

Comment:
Could you please try building this into the existing Dockerfile? It's pretty much the same, except for a few ARM modifications, so keeping the two Dockerfile in sync would be quite the maintenance burden. There are some docs on multi-platform builds here: https://docs.docker.com/build/building/multi-platform/
The available variables are documented here: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

Reply:
Here's a relevant guide: https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/